controllers/front/Ajax.php

```diff
@@ -107,8 +108,25 @@ public function postProcess()
                 break;
             case 'updateStreetSelect':
                 $city = Tools::getValue('city');
-                $countryCode = Configuration::get(Config::WEB_SERVICE_COUNTRY);
                 $this->updateStreetSelect($countryCode, $city);
+                break;
+            case 'saveSelectedPhoneNumber':
+                /** @var  \Invertus\dpdBaltics\Service\CarrierPhoneService $phoneService */
+                $phoneService = $this->module->getContainer(\Invertus\dpdBaltics\Service\CarrierPhoneService::class);
+
+                if(!$phoneService->saveCarrierPhone($cartId,
+                    Tools::getValue('phone_number'),
+                    Tools::getValue('phone_area')
```

Comment:
does the service inisde has phone number validation ? Do you validate it on back-end side, if so , how ? If not, then I suppose you should do that.

Reply:
Added validation inside function